config/vufind/CookieConsent.yaml

```diff
@@ -0,0 +1,141 @@
+#     Expiration        Expiration time for the cookie (number or "session")
+#     ExpirationUnit    Unit of expiration for translation when Expiration is a
+#                       numeric value. Out of the box supported values are "days" and
+#                       "months".
+#     ThirdParty        Whether the cookie comes from a third party (true/false,
+#                       default is false)
+#
+#   AutoClearCookies  A list of cookies that should be automatically removed if
+#                     consent for a category is withdrawn:
+#
+#     Name              A regular expression to match cookie names that should be
+#                       cleared
+#
+# Iframe service configuration (see the VuFind wiki for an example:
+# https://vufind.org/wiki/configuration:cookie_consent)
+#
+#   EmbedUrl          Template for embedded content url. {data-id} can be used as a
+#                     placeholder for the data-id attribute of the tag.
+#   ThumbnailUrl      Template for thumbnail image url. {data-id} can be used as a
+#                     placeholder for the data-id attribute of the tag.
+#   IframeAllow       allow attribute for the iframe
+#   Description       Description text displayed alongside buttons to display the
+#                     content
+#   AllowOnceText     Text for the "Allow once" button
+#   AllowAlwaysText   Text for the "Allow always" button (remembers the choice)
+#
+# You can use e.g. https://cookiedatabase.org/ to find out the cookies used by most
+# popular services.
+#
+# All textual strings are translatable. The following placeholders can be used in
+# settings and translated texts:
+#
+# {{consent_cookie_name}}         Name of the consent cookie (default cc_cookie)
+# {{consent_cookie_expiration}}   Consent cookie expiration time
+# {{current_host_name}}           Server name of the VuFind server
+# {{vufind_cookie_domain}}        Domain for VuFind's cookies
+# {{vufind_session_cookie}}       VuFind session cookie name
+#
+CookieConsent:
+  #CookieExpiration: 365
+  Categories:
+    essential:
+      Title: CookieConsent::essential_cookies_title_html
+      Description: CookieConsent::essential_cookies_description_html
+      DefaultEnabled: true
+      Essential: true
+      Cookies:
+        - Name: '{{consent_cookie_name}}'
+          Domain: '{{vufind_cookie_domain}}'
+          Description: CookieConsent::cookie_description_consent_html
+          Expiration: '{{consent_cookie_expiration}}'
+          ExpirationUnit: days
+        - Name: 'cc_*'
+          Domain: '{{vufind_cookie_domain}}'
+          Description: CookieConsent::cookie_description_consent_html
+          Expiration: '{{consent_cookie_expiration}}'
+          ExpirationUnit: days
+        - Name: language
+          Domain: '{{vufind_cookie_domain}}'
+          Description: CookieConsent::cookie_description_language_html
+          Expiration: session
+        - Name: sidefacet-* (session storage)
```

Comment:
I'm a little confused by the `(session storage)` here. I was assuming that the names were being used to actually match up cookies for functional reasons. Does that mean that `(session storage)` has a specific meaning that should be documented? Or is this field purely descriptive (in which case, how are the actual cookies matched up?)

Reply:
Name in the Cookies array is just descriptive. I'll add a note. It's only in `AutoClearCookies` where it has functionality.